src/hotspot/cpu/x86/sharedRuntime_x86_64.cpp

```diff
@@ -199,7 +217,32 @@ OopMap* RegisterSaver::save_live_registers(MacroAssembler* masm, int additional_
   // to be under the return like a normal enter.
 
   __ enter();          // rsp becomes 16-byte aligned here
-  __ push_CPU_state(); // Push a multiple of 16 bytes
+  __ pushf();
+  // Make sure rsp stays 16-byte aligned
+  __ subq(rsp, 8);
+  // Push CPU state in multiple of 16 bytes
+  __ subq(rsp, 16 * wordSize);
+
+  __ movq(Address(rsp, 15 * wordSize), rax);
```

Comment:
Consider moving saving register code into separate method. You have 3 places where you use it.